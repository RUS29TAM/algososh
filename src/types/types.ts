import {ElementStates} from "./element-states";

export interface ICircleDetail {
    symbol?: null | string;
    state: ElementStates;
}

export interface IColumn {
    number: number;
    state: ElementStates;
}
