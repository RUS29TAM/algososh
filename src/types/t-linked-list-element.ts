export type TLinkedListElement = {
    isAddHead: boolean;
    isDeleteHead: boolean;
    isAddTail: boolean;
    isDeleteTail: boolean;
    isAddByIndex: boolean;
    isDeleteByIndex: boolean;
}
