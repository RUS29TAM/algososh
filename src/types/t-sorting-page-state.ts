export type TSortingPageState = {
    isIncreasing: boolean;
    isDecreasing: boolean;
    isLoading: boolean;
}
