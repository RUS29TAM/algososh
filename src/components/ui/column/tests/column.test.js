import renderer from "react-test-renderer";
import {ElementStates} from "../../../../types/element-states";
import {Column} from "../column";

describe('Тест для компонента Column', () => {
    it('Компонент Column отрисовка по индексу - тест пройден', () => {
        const column = renderer.create(<Column index={0} />).toJSON();
        expect(column).toMatchSnapshot();
    })

    it('Компонент Column отрисовка отрисовка в состоянии Default - тест пройден', () => {
        const column = renderer.create(<Column index={1} state={ElementStates.Default} />).toJSON();
        expect(column).toMatchSnapshot();
    })

    it('Компонент Column отрисовка отрисовка в состоянии Changing - тест пройден', () => {
        const column = renderer.create(<Column index={2} state={ElementStates.Changing} />).toJSON();
        expect(column).toMatchSnapshot();
    })

    it('Компонент Column отрисовка отрисовка в состоянии Modified - тест пройден', () => {
        const column = renderer.create(<Column index={3} state={ElementStates.Modified} />).toJSON();
        expect(column).toMatchSnapshot();
    })
})