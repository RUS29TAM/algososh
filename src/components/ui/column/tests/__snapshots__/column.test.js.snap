// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест для компонента Column Компонент Column отрисовка отрисовка в состоянии Changing - тест пройден 1`] = `
<div
  className="content "
>
  <div
    className="column changing"
    style={
      Object {
        "height": 6.4,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  >
    2
  </p>
</div>
`;

exports[`Тест для компонента Column Компонент Column отрисовка отрисовка в состоянии Default - тест пройден 1`] = `
<div
  className="content "
>
  <div
    className="column default"
    style={
      Object {
        "height": 3.2,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  >
    1
  </p>
</div>
`;

exports[`Тест для компонента Column Компонент Column отрисовка отрисовка в состоянии Modified - тест пройден 1`] = `
<div
  className="content "
>
  <div
    className="column modified"
    style={
      Object {
        "height": 9.6,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  >
    3
  </p>
</div>
`;

exports[`Тест для компонента Column Компонент Column отрисовка по индексу - тест пройден 1`] = `
<div
  className="content "
>
  <div
    className="column default"
    style={
      Object {
        "height": 1,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  >
    0
  </p>
</div>
`;
